src/robotlibcore.py

```diff
@@ -101,36 +101,59 @@ def get_keyword_arguments(self, name):
         kw_method = self.__get_keyword(name)
         if kw_method is None:
             return None
-        args, defaults, varargs, kwargs = self.__get_arg_spec(kw_method)
-        if robot_version >= '3.2':
-            args += self.__new_default_spec(defaults)
-        else:
+        args, defaults, varargs, kwargs, kwonlydefaults = self.__get_arg_spec(kw_method)
+        if self.__rf_31:
             args += self.__old_default_spec(defaults)
+        else:
+            args += self.__new_default_spec(defaults)
         if varargs:
             args.append('*%s' % varargs)
         if kwargs:
             args.append('**%s' % kwargs)
+        if kwonlydefaults:
+            args += self.__kwonlydefaults_spec(kwonlydefaults)
         return args
 
+    @property
+    def __rf_31(self):
+        return robot_version < '3.2'
```

Comment:
This could be a module level constant like `PY2`:

    RF31 = robot_version < '3.2'

Perhaps this could be defined as well:

    RF32 = not RF31